agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -792,8 +792,33 @@ bool backhaul_manager::backhaul_fsm_main(bool &skip_select)
         break;
     }
     case EState::MASTER_DISCOVERY: {
-
         auto db = AgentDB::get();
+
+        // --- begin of certificate mode-related checks
+
+        // In certification mode we want to wait till dev_set_config is received (wired backhaul)
+        // or start_wps_registration (wireless backhaul).
+        auto certification_mode = bpl::cfg_get_certification_mode();
+        if (certification_mode < 0) {
+            LOG(ERROR) << "Failed reading certification_mode: " << certification_mode;
+            return false;
+        }
+
+        bool wired_backhaul = db->backhaul.connection_type == AgentDB::sBackhaul::eConnectionType::Wired;
+
+        if (certification_mode && wired_backhaul && !is_main_agent()) {
```

Comment:
Please rebase to #1590 and take the `certification_mode` directly from the DB. It will save the error handling and you could use it directly on the condition.